lib/puppet/parser/environment_compiler.rb

```diff
@@ -82,4 +136,12 @@ def evaluate_applications
       end
     end
   end
+
+  # Overrides the regular compiler to be able to return the list of resources after a prune
+  # has taken place in the graph representation. Before a prune, the list is the same as in the regular
+  # compiler
+  #
+  def resources
+    @pruned_resources || super
```

Comment:
Perhaps not ideal to have a attribute named @resources and an accessor method with the same name that doesn't return that attribute. Are both @resource and @pruned_resources needed or could the prune_catalog replace the former?